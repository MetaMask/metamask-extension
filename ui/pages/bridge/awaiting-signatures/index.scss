.awaiting-bridge-signatures {
  display: flex;
  flex-flow: column;
  align-items: center;
  flex: 1;
  width: 100%;
  margin-top: 40%;

  &__content {
    flex-flow: column;
  }

  div {
    text-align: center;
    display: flex;
    justify-content: center;
  }

  &__steps {
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    margin: 16px auto 12px auto;

    li {
      margin-bottom: 4px;
      display: flex;
      align-items: center;
    }
  }
}
