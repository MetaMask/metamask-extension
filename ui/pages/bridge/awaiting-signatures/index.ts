export { default } from './awaiting-signatures';
