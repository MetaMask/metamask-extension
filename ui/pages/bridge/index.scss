@use "design-system";

@import 'prepare/index';
@import 'quotes/index';


.bridge {
  max-height: 100vh;
  width: 360px;
  position: relative;

  &__container {
    width: 100%;

    .multichain-page-footer {
      position: absolute;
      width: 100%;
      height: 80px;
      bottom: 0;
      padding: 16px;
      display: flex;

      button {
        flex: 1;
        height: 100%;
        font-size: 14px;
        font-weight: 500;
      }
    }
  }
}
