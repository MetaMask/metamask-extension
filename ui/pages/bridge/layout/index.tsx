import Column from './column';
import Row from './row';
import Tooltip from './tooltip';

export { Column, Row, Tooltip };
