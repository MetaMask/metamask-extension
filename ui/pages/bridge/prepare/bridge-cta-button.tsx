import React, { useMemo } from 'react';
import { useSelector } from 'react-redux';
import { Button } from '../../../components/component-library';
import {
  getBridgeQuotes,
  getFromAmount,
  getFromChain,
  getFromToken,
  getToAmount,
  getToChain,
  getToToken,
} from '../../../ducks/bridge/selectors';
import { useI18nContext } from '../../../hooks/useI18nContext';

export const BridgeCTAButton = () => {
  const t = useI18nContext();
  const fromToken = useSelector(getFromToken);
  const toToken = useSelector(getToToken);

  const fromChain = useSelector(getFromChain);
  const toChain = useSelector(getToChain);

  const fromAmount = useSelector(getFromAmount);
  const toAmount = useSelector(getToAmount);

  const { isLoading } = useSelector(getBridgeQuotes);

  const isTxSubmittable =
    fromToken && toToken && fromChain && toChain && fromAmount && toAmount;

  const label = useMemo(() => {
    if (isLoading && !isTxSubmittable) {
      return t('swapFetchingQuotes');
    }

    if (!fromAmount) {
      if (!toToken) {
        return t('bridgeSelectTokenAndAmount');
      }
      return t('bridgeEnterAmount');
    }

    if (isTxSubmittable) {
      return t('confirm');
    }

    return t('swapSelectToken');
  }, [isLoading, fromAmount, toToken, isTxSubmittable]);

  return (
    <Button
      data-testid="bridge-cta-button"
      onClick={() => {
        if (isTxSubmittable) {
          // dispatch tx submission
        }
      }}
      disabled={!isTxSubmittable}
    >
      {label}
    </Button>
  );
};
