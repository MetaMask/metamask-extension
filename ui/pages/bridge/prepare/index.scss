@use "design-system";

.tokens-main-view-modal {
  .multichain-asset-picker-list-item .mm-badge-wrapper__badge-container {
    display: none;
  }
}

.prepare-bridge-page {
  display: flex;
  flex-flow: column;
  flex: 1;
  width: 100%;
  gap: 24px;

  &__content {
    display: flex;
    flex-direction: column;
    padding: 16px 0 16px 0;
    border-radius: 8px;
    border: 1px solid var(--color-border-muted);
  }

  .bridge-box {
    display: flex;
    flex-direction: column;
    gap: 4px;
    justify-content: center;
    padding: 8px 16px 8px 16px;
  }

  &__input-row {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    width: 298px;
    gap: 16px;

    input[type="number"]::-webkit-inner-spin-button {
      -webkit-appearance: none;
      -moz-appearance: none;
      display: none;
    }

    input[type="number"]:hover::-webkit-inner-spin-button {
      -webkit-appearance: none;
      -moz-appearance: none;
      display: none;
    }

    .mm-text-field {
      background-color: inherit;

      &--focused {
        outline: none;
      }
    }

    .defined {
      opacity: 1;

      & > .mm-input--disabled {
        opacity: 1;
      }
    }
  }

  .amount-input {
    border: none;

    input {
      text-align: right;
      padding-right: 0;
      font-size: 24px;
      font-weight: 700;

      &:focus,
      &:focus-visible {
        outline: none;
      }
    }

    .mm-text-field--focused {
      outline: none;
    }
  }

  &__amounts-row {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    width: 298px;
    height: 22px;

    p,
    span {
      color: var(--color-text-alternative);
      font-size: 12px;
    }
  }

  .asset-picker {
    border: 1px solid var(--color-border-muted);
    height: 40px;
    min-width: fit-content;
    max-width: fit-content;
    background-color: inherit;

    p {
      font-size: 14px;
      font-weight: 500;
    }

    .mm-avatar-token {
      height: 24px;
      width: 24px;
      border: 1px solid var(--color-border-muted);
    }

    .mm-badge-wrapper__badge-container .mm-avatar-base {
      height: 10px;
      width: 10px;
      border: none;
    }
  }

  &__switch-tokens {
    display: flex;
    justify-content: center;
    align-items: center;

    &::before,
    &::after {
      content: '';
      border-top: 1px solid var(--color-border-muted);
      flex-grow: 1;
    }

    button {
      border-radius: 50%;
      padding: 10px;
      border: 1px solid var(--color-border-muted);
      transition: all 0.3s ease-in-out;
      cursor: pointer;
      width: 40px;
      height: 40px;

      &:hover:enabled {
        background: var(--color-background-default-hover);

        .mm-icon {
          color: var(--color-icon-default);
        }
      }

      &:active {
        background: var(--color-background-default-pressed);

        .mm-icon {
          color: var(--color-icon-default);
        }
      }

      .rotate {
        transform: rotate(360deg);
      }
    }

    .mm-icon {
      color: var(--color-icon-alternative);
      transition: all 0.3s ease-in-out;
    }

    &:disabled {
      cursor: not-allowed;
    }
  }
}
