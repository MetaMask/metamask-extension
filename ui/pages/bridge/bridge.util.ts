import { Hex, add0x } from '@metamask/utils';
import {
  BridgeFeatureFlagsKey,
  BridgeFeatureFlags,
  // TODO: Remove restricted import
  // eslint-disable-next-line import/no-restricted-paths
} from '../../../app/scripts/controllers/bridge/types';
import {
  BRIDGE_API_BASE_URL,
  BRIDGE_CLIENT_ID,
} from '../../../shared/constants/bridge';
import { MINUTE } from '../../../shared/constants/time';
import fetchWithCache from '../../../shared/lib/fetch-with-cache';
import {
  decimalToHex,
  hexToDecimal,
} from '../../../shared/modules/conversion.utils';
import {
  SWAPS_CHAINID_DEFAULT_TOKEN_MAP,
  SwapsTokenObject,
} from '../../../shared/constants/swaps';
import {
  isSwapsDefaultTokenAddress,
  isSwapsDefaultTokenSymbol,
} from '../../../shared/modules/swaps.utils';
// TODO: Remove restricted import
// eslint-disable-next-line import/no-restricted-paths
import { REFRESH_INTERVAL_MS } from '../../../app/scripts/controllers/bridge/constants';
import {
  BridgeAsset,
  BridgeFlag,
  FeatureFlagResponse,
  FeeData,
  FeeType,
  Quote,
  QuoteRequest,
  QuoteResponse,
  TxData,
} from './types';
import {
  FEATURE_FLAG_VALIDATORS,
  QUOTE_VALIDATORS,
  TX_DATA_VALIDATORS,
  TOKEN_VALIDATORS,
  validateResponse,
  QUOTE_RESPONSE_VALIDATORS,
  FEE_DATA_VALIDATORS,
} from './utils/validators';

const CLIENT_ID_HEADER = { 'X-Client-Id': BRIDGE_CLIENT_ID };
const CACHE_REFRESH_TEN_MINUTES = 10 * MINUTE;

export async function fetchBridgeFeatureFlags(): Promise<BridgeFeatureFlags> {
  const url = `${BRIDGE_API_BASE_URL}/getAllFeatureFlags`;
  const rawFeatureFlags = await fetchWithCache({
    url,
    fetchOptions: { method: 'GET', headers: CLIENT_ID_HEADER },
    cacheOptions: { cacheRefreshTime: CACHE_REFRESH_TEN_MINUTES },
    functionName: 'fetchBridgeFeatureFlags',
  });

  if (
    validateResponse<FeatureFlagResponse>(
      FEATURE_FLAG_VALIDATORS,
      rawFeatureFlags,
      url,
    )
  ) {
    return {
      [BridgeFeatureFlagsKey.EXTENSION_CONFIG]:
        rawFeatureFlags[BridgeFlag.EXTENSION_CONFIG],
      [BridgeFeatureFlagsKey.EXTENSION_SUPPORT]:
        rawFeatureFlags[BridgeFlag.EXTENSION_SUPPORT],
      [BridgeFeatureFlagsKey.NETWORK_SRC_ALLOWLIST]: rawFeatureFlags[
        BridgeFlag.NETWORK_SRC_ALLOWLIST
      ].map((chainIdDec) => add0x(decimalToHex(chainIdDec))),
      [BridgeFeatureFlagsKey.NETWORK_DEST_ALLOWLIST]: rawFeatureFlags[
        BridgeFlag.NETWORK_DEST_ALLOWLIST
      ].map((chainIdDec) => add0x(decimalToHex(chainIdDec))),
    };
  }

  return {
    [BridgeFeatureFlagsKey.EXTENSION_CONFIG]: {
      refreshRate: REFRESH_INTERVAL_MS,
      maxRefreshCount: 5,
    },
    // TODO set default to true once bridging is live
    [BridgeFeatureFlagsKey.EXTENSION_SUPPORT]: false,
    // TODO set default to ALLOWED_BRIDGE_CHAIN_IDS once bridging is live
    [BridgeFeatureFlagsKey.NETWORK_SRC_ALLOWLIST]: [],
    // TODO set default to ALLOWED_BRIDGE_CHAIN_IDS once bridging is live
    [BridgeFeatureFlagsKey.NETWORK_DEST_ALLOWLIST]: [],
  };
}

// Returns a list of enabled (unblocked) tokens
export async function fetchBridgeTokens(
  chainId: Hex,
): Promise<Record<string, SwapsTokenObject>> {
  // TODO make token api v2 call
  const url = `${BRIDGE_API_BASE_URL}/getTokens?chainId=${hexToDecimal(
    chainId,
  )}`;
  const tokens = await fetchWithCache({
    url,
    fetchOptions: { method: 'GET', headers: CLIENT_ID_HEADER },
    cacheOptions: { cacheRefreshTime: CACHE_REFRESH_TEN_MINUTES },
    functionName: 'fetchBridgeTokens',
  });

  const nativeToken =
    SWAPS_CHAINID_DEFAULT_TOKEN_MAP[
      chainId as keyof typeof SWAPS_CHAINID_DEFAULT_TOKEN_MAP
    ];

  const transformedTokens: Record<string, SwapsTokenObject> = {};
  if (nativeToken) {
    transformedTokens[nativeToken.address] = nativeToken;
  }

  tokens.forEach((token: unknown) => {
    if (
      validateResponse<SwapsTokenObject>(TOKEN_VALIDATORS, token, url) &&
      !(
        isSwapsDefaultTokenSymbol(token.symbol, chainId) ||
        isSwapsDefaultTokenAddress(token.address, chainId)
      )
    ) {
      transformedTokens[token.address] = token;
    }
  });
  return transformedTokens;
}

// Returns a list of bridge tx quotes
export async function fetchBridgeQuotes(
  request: QuoteRequest,
): Promise<QuoteResponse[]> {
  const queryParams = new URLSearchParams({
    walletAddress: request.walletAddress,
    srcChainId: request.srcChainId.toString(),
    destChainId: request.destChainId.toString(),
    srcTokenAddress: request.srcTokenAddress,
    destTokenAddress: request.destTokenAddress,
    srcTokenAmount: request.srcTokenAmount,
    slippage: request.slippage.toString(),
    insufficientBal: request.insufficientBal ? 'true' : 'false',
    resetApproval: request.resetApproval ? 'true' : 'false',
  });
  const url = `${BRIDGE_API_BASE_URL}/getQuote?${queryParams}`;
  const quotes = await fetchWithCache({
    url,
    fetchOptions: { method: 'GET', headers: CLIENT_ID_HEADER },
    cacheOptions: { cacheRefreshTime: 0 },
    functionName: 'fetchBridgeQuotes',
  });

  const filteredQuotes = quotes.filter((quoteResponse: QuoteResponse) => {
    const { quote, approval, trade } = quoteResponse;
    return (
      validateResponse<QuoteResponse>(
        QUOTE_RESPONSE_VALIDATORS,
        quoteResponse,
        url,
      ) &&
      validateResponse<Quote>(QUOTE_VALIDATORS, quote, url) &&
      validateResponse<BridgeAsset>(TOKEN_VALIDATORS, quote.srcAsset, url) &&
      validateResponse<BridgeAsset>(TOKEN_VALIDATORS, quote.destAsset, url) &&
      validateResponse<TxData>(TX_DATA_VALIDATORS, trade, url) &&
      validateResponse<FeeData>(
        FEE_DATA_VALIDATORS,
        quote.feeData[FeeType.METABRIDGE],
        url,
      ) &&
      (approval
        ? validateResponse<TxData>(TX_DATA_VALIDATORS, approval, url)
        : true)
    );
  });
  return filteredQuotes;
}
