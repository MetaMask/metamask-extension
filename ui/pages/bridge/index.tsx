import React, { useContext, useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Switch, useHistory } from 'react-router-dom';
import { I18nContext } from '../../contexts/i18n';
import { clearSwapsState } from '../../ducks/swaps/swaps';
import {
  DEFAULT_ROUTE,
  SWAPS_MAINTENANCE_ROUTE,
  PREPARE_SWAP_ROUTE,
  CROSS_CHAIN_SWAP_ROUTE,
} from '../../helpers/constants/routes';
import { resetBackgroundSwapsState } from '../../store/actions';
import FeatureToggledRoute from '../../helpers/higher-order-components/feature-toggled-route';
import {
  ButtonIcon,
  ButtonIconSize,
  IconName,
} from '../../components/component-library';
import { getIsBridgeChain, getIsBridgeEnabled } from '../../selectors';
import useBridging from '../../hooks/bridge/useBridging';
import {
  Content,
  Footer,
  Header,
} from '../../components/multichain/pages/page';
import { getProviderConfig } from '../../ducks/metamask/metamask';
import { resetInputFields, setFromChain } from '../../ducks/bridge/actions';
import PrepareBridgePage from './prepare/prepare-bridge-page';
import { BridgeCTAButton } from './prepare/bridge-cta-button';

const CrossChainSwap = () => {
  const t = useContext(I18nContext);

  useBridging();

  const history = useHistory();
  const dispatch = useDispatch();

  const isBridgeEnabled = useSelector(getIsBridgeEnabled);
  const providerConfig = useSelector(getProviderConfig);
  const isBridgeChain = useSelector(getIsBridgeChain);

  useEffect(() => {
    isBridgeChain &&
      isBridgeEnabled &&
      providerConfig &&
      dispatch(setFromChain(providerConfig.chainId));

    return () => {
      dispatch(resetInputFields());
    };
  }, [isBridgeChain, isBridgeEnabled, providerConfig]);

  const redirectToDefaultRoute = async () => {
    history.push({
      pathname: DEFAULT_ROUTE,
      state: { stayOnHomePage: true },
    });
    dispatch(clearSwapsState());
    await dispatch(resetBackgroundSwapsState());
  };

  return (
    <div className="bridge">
      <div className="bridge__container">
        <Header
          className="bridge__header"
          startAccessory={
            <ButtonIcon
              iconName={IconName.ArrowLeft}
              size={ButtonIconSize.Sm}
              ariaLabel={t('back')}
              onClick={redirectToDefaultRoute}
            />
          }
          endAccessory={
            <ButtonIcon
              iconName={IconName.Setting}
              size={ButtonIconSize.Sm}
              ariaLabel={t('settings')}
            />
          }
        >
          {t('bridge')}
        </Header>
        <Content className="bridge__content">
          <Switch>
            <FeatureToggledRoute
              redirectRoute={SWAPS_MAINTENANCE_ROUTE}
              flag={isBridgeEnabled}
              path={CROSS_CHAIN_SWAP_ROUTE + PREPARE_SWAP_ROUTE}
              render={() => {
                return <PrepareBridgePage />;
              }}
            />
          </Switch>
        </Content>
        <Footer>
          <BridgeCTAButton />
        </Footer>
      </div>
    </div>
  );
};

export default CrossChainSwap;
