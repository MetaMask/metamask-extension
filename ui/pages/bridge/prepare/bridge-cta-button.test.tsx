import React from 'react';
import { renderWithProvider } from '../../../../test/jest';
import configureStore from '../../../store/store';
import { createBridgeMockStore } from '../../../../test/jest/mock-store';
import { CHAIN_IDS } from '../../../../shared/constants/network';
import mockBridgeQuotesNativeErc20 from '../../../../test/data/bridge/mock-quotes-native-erc20.json';
// TODO: Remove restricted import
// eslint-disable-next-line import/no-restricted-paths
import { RequestStatus } from '../../../../app/scripts/controllers/bridge/constants';
import { BridgeCTAButton } from './bridge-cta-button';

describe('BridgeCTAButton', () => {
  it("should render the component's initial state", () => {
    const mockStore = createBridgeMockStore(
      {
        srcNetworkAllowlist: [CHAIN_IDS.MAINNET, CHAIN_IDS.OPTIMISM],
        destNetworkAllowlist: [CHAIN_IDS.OPTIMISM],
      },
      { fromTokenInputValue: 1 },
    );
    const { container, getByText, getByRole } = renderWithProvider(
      <BridgeCTAButton />,
      configureStore(mockStore),
    );

    expect(container).toMatchSnapshot();

    expect(getByText('Select token')).toBeInTheDocument();
    expect(getByRole('button')).toBeDisabled();
  });

  it('should render the component when amount is missing', () => {
    const mockStore = createBridgeMockStore(
      {
        srcNetworkAllowlist: [CHAIN_IDS.MAINNET, CHAIN_IDS.OPTIMISM],
        destNetworkAllowlist: [CHAIN_IDS.LINEA_MAINNET],
      },
      {
        fromTokenInputValue: null,
        fromToken: 'ETH',
        toToken: 'ETH',
        toChainId: CHAIN_IDS.LINEA_MAINNET,
      },
      {},
    );
    const { getByText, getByRole } = renderWithProvider(
      <BridgeCTAButton />,
      configureStore(mockStore),
    );

    expect(getByText('Enter amount')).toBeInTheDocument();
    expect(getByRole('button')).toBeDisabled();
  });

  it('should render the component when amount and dest token is missing', () => {
    const mockStore = createBridgeMockStore(
      {
        srcNetworkAllowlist: [CHAIN_IDS.MAINNET, CHAIN_IDS.OPTIMISM],
        destNetworkAllowlist: [CHAIN_IDS.LINEA_MAINNET],
      },
      {
        fromTokenInputValue: null,
        fromToken: 'ETH',
        toToken: null,
        toChainId: CHAIN_IDS.LINEA_MAINNET,
      },
      {},
    );
    const { getByText, getByRole } = renderWithProvider(
      <BridgeCTAButton />,
      configureStore(mockStore),
    );

    expect(getByText('Select token and amount')).toBeInTheDocument();
    expect(getByRole('button')).toBeDisabled();
  });

  it('should render the component when tx is submittable', () => {
    const mockStore = createBridgeMockStore(
      {
        srcNetworkAllowlist: [CHAIN_IDS.MAINNET, CHAIN_IDS.OPTIMISM],
        destNetworkAllowlist: [CHAIN_IDS.LINEA_MAINNET],
      },
      {
        fromTokenInputValue: 1,
        fromToken: 'ETH',
        toToken: 'ETH',
        toChainId: CHAIN_IDS.LINEA_MAINNET,
      },
      {
        quotes: mockBridgeQuotesNativeErc20,
        quotesLastFetched: Date.now(),
        quotesLoadingStatus: RequestStatus.FETCHED,
      },
    );
    const { getByText, getByRole } = renderWithProvider(
      <BridgeCTAButton />,
      configureStore(mockStore),
    );

    expect(getByText('Confirm')).toBeInTheDocument();
    expect(getByRole('button')).not.toBeDisabled();
  });

  it('should disable the component when quotes are loading and there are no existing quotes', () => {
    const mockStore = createBridgeMockStore(
      {
        srcNetworkAllowlist: [CHAIN_IDS.MAINNET, CHAIN_IDS.OPTIMISM],
        destNetworkAllowlist: [CHAIN_IDS.LINEA_MAINNET],
      },
      {
        fromTokenInputValue: 1,
        fromToken: 'ETH',
        toToken: 'ETH',
        toChainId: CHAIN_IDS.LINEA_MAINNET,
      },
      {
        quotes: [],
        quotesLastFetched: Date.now(),
        quotesLoadingStatus: RequestStatus.LOADING,
      },
    );
    const { getByText, getByRole } = renderWithProvider(
      <BridgeCTAButton />,
      configureStore(mockStore),
    );

    expect(getByText('Fetching quotes...')).toBeInTheDocument();
    expect(getByRole('button')).toBeDisabled();
  });

  it('should enable the component when quotes are loading and there are existing quotes', () => {
    const mockStore = createBridgeMockStore(
      {
        srcNetworkAllowlist: [CHAIN_IDS.MAINNET, CHAIN_IDS.OPTIMISM],
        destNetworkAllowlist: [CHAIN_IDS.LINEA_MAINNET],
      },
      {
        fromTokenInputValue: 1,
        fromToken: 'ETH',
        toToken: 'ETH',
        toChainId: CHAIN_IDS.LINEA_MAINNET,
      },
      {
        quotes: mockBridgeQuotesNativeErc20,
        quotesLastFetched: Date.now(),
        quotesLoadingStatus: RequestStatus.LOADING,
      },
    );
    const { getByText, getByRole } = renderWithProvider(
      <BridgeCTAButton />,
      configureStore(mockStore),
    );

    expect(getByText('Confirm')).toBeInTheDocument();
    expect(getByRole('button')).not.toBeDisabled();
  });
});
