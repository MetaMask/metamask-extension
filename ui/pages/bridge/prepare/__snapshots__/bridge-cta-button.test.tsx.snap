// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BridgeCTAButton should render the component's initial state 1`] = `
<div>
  <button
    class="mm-box mm-text mm-button-base mm-button-base--size-md mm-button-base--disabled mm-button-primary mm-button-primary--disabled mm-text--body-md-medium mm-box--padding-0 mm-box--padding-right-4 mm-box--padding-left-4 mm-box--display-inline-flex mm-box--justify-content-center mm-box--align-items-center mm-box--color-primary-inverse mm-box--background-color-primary-default mm-box--rounded-pill"
    data-testid="bridge-cta-button"
    data-theme="light"
    disabled=""
  >
    Select token
  </button>
</div>
`;
